### PR TITLE
cmek split 2/8: TiKV encryption metadata HTTP client

### PR DESCRIPTION
**Pull request by tenfyzhong**
### What problem does this PR solve?

Issue Number: ref #3943

This is split PR **2/8** from #3955.

### What is changed and how it works?

This PR adds TiKV-side encryption metadata retrieval:

- Add TiKV encryption metadata HTTP client.
- Add JSON/bytes decoding helpers for metadata payloads.
- Parse and normalize metadata for keyspace-level encryption usage.
- Add unit tests for endpoint behavior and decoding logic.

This PR focuses only on metadata retrieval and parsing, and does not yet wire server lifecycle.

#### Tests

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

##### Will it cause performance regression or break compatibility?

No direct user-facing behavior change in this split; it introduces a client used by later PRs.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note

```release-note
None
```

## Summary by CodeRabbit

* **New Features**
  * Added encryption metadata client for TiKV store integration with support for protobuf encoding and HTTP communication with configurable TLS.

* **Tests**
  * Added comprehensive test coverage for encryption metadata retrieval, including edge cases, error handling, and protocol validation.

### CODE REVIEW

File: pkg/encryption/tikv_http_client.go [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+			zap.String("statusAddr", statusAddr),
+			zap.Uint32("keyspaceID", keyspaceID),
+			zap.String("url", storeURL),
+			zap.String("contentType", contentType),
+			zap.Int("bodySize", len(body)),
+			zap.Error(jsonErr))
+
+		metaResp, err = decodeEncryptionMetaResponseFromProtobuf(body)
+		if err != nil {
+			decodeErr := errors.Annotatef(err, "json decode failed: %v", jsonErr)
```
Comment (3AceShowHand) [inline pkg/encryption/tikv_http_client.go]: Annotate is not recommanded, I always use errors.WrapError(ErrDecodeFailed, err)

**Review by 3AceShowHand — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]
